### PR TITLE
Csi 1167 handle device upgrade errors

### PR DESCRIPTION
**Pull request by HarishRocks**

### CODE REVIEW

File: cysync/src/store/hooks/flows/useDeviceUpgrade.ts [hunk 1]
```diff
@@ -568,8 +571,6 @@ export const useDeviceUpgrade: UseDeviceUpgrade = (isInitial?: boolean) => {
         setErrorResolutionState(
           DeviceUpgradeErrorResolutionState.DEVICE_AUTH_REQUIRED
         );
-        const cyError = new CyError(CysyncError.DEVICE_AUTH_FAILED);
-        setErrorObj(handleErrors(errorObj, cyError, flowName));
```
Comment (irshadCypherock) [inline cysync/src/store/hooks/flows/useDeviceUpgrade.ts:572]: We're not setting error here?

File: cysync/src/errors/helpers.ts [hunk 2]
```diff
@@ -9,16 +9,21 @@ import logger from '../utils/logger';
...
   currError: CyError,
   err: CyError,
   flow?: string,
   metadata?: any
-) => {
+): CyError => {
   //TODO:  handle cascade effect properly
   if (currError.isSet) {
+    if (isUnknownError(err.getCode()) || currError.compare(err))
+      return currError;
```
Comment (irshadCypherock) [inline cysync/src/errors/helpers.ts:25]: Need more explanation for this

File: cysync/src/store/hooks/flows/useDeviceAuth.ts [hunk 3]
```diff
@@ -105,7 +105,12 @@ export const useDeviceAuth: UseDeviceAuth = isInitial => {
       process.env.BUILD_TYPE === 'debug' &&
       localStorage.getItem('disableProvision') === 'true';
 
-    setConfirmed(1);
+    if (inTestApp) {
+      setConfirmed(2);
+      setVerified(1);
+    } else {
+      setConfirmed(1);
+    }
```
Comment (irshadCypherock) [inline cysync/src/store/hooks/flows/useDeviceAuth.ts:113]: I think this already happens from the `protocol` module

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: cysync/src/store/hooks/flows/useDeviceUpgrade.ts (hunk 1 above)
Comment (HarishRocks) [reply cysync/src/store/hooks/flows/useDeviceUpgrade.ts:572]: Not necessary, on device auth failure. The errors are set by `useDeviceAuth` hook itself. So if there's any auth error, `useDeviceUpgrade` is expected to just forward it.

File: cysync/src/errors/error.ts [hunk 4]
```diff
@@ -71,6 +71,10 @@ class CyError extends DisplayError {
     this.childErrors.push(dispError);
     return dispError;
   }
+
+  public compare(err: CyError) {
+    return this.getCode() === err.getCode();
+  }
```
[the remark this answers is not in the document]
Comment (HarishRocks) [reply cysync/src/errors/error.ts:77]: going with `isEqualTo`

File: cysync/src/errors/helpers.ts (hunk 2 above)
Comment (HarishRocks) [reply cysync/src/errors/helpers.ts:25]: added some comments

File: cysync/src/errors/helpers.ts [hunk 5]
```diff
@@ -41,7 +46,23 @@ const handleErrors = (
...
+      DeviceErrorType.FIRMWARE_SIZE_LIMIT_EXCEEDED,
+      DeviceErrorType.WRONG_HARDWARE_VERSION,
+      DeviceErrorType.WRONG_MAGIC_NUMBER,
+      DeviceErrorType.SIGNATURE_NOT_VERIFIED,
+      DeviceErrorType.LOWER_FIRMWARE_VERSION
+    ].includes(err.errorType)
+  ) {
+    cyError.setError(CysyncError.DEVICE_UPGRADE_KNOWN_ERROR);
+  }
+};
```
[the remark this answers is not in the document]
Comment (HarishRocks) [reply cysync/src/errors/helpers.ts:62]: Actually we can, it wouldn't really be needed on `handleDeviceErrors` . But I don't see any problem with that

File: cysync/src/store/hooks/flows/useDeviceAuth.ts (hunk 3 above)
Comment (HarishRocks) [reply cysync/src/store/hooks/flows/useDeviceAuth.ts:113]: It wasn't happening, So I had to. Could you recheck?

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**